### ISSUE TITLE
Python sklearn XGBClassifier cannot used in plot_partial_dependence

### ISSUE THREAD

**Issue by Brill-Wang** [dmlc/xgboost#2035]
xgbModel = xgb.XGBClassifier(max_depth=5, learning_rate=0.3, silent=1, objective='binary:logistic', 
                  min_child_weight=10, gamma=0.1, subsample=0.8, scale_pos_weight=111.5).fit(feat, labels)

fig, axs = plot_partial_dependence(xgbModel, feat, focus_features,
                                       feature_names=featurenames,
                                       n_jobs=3, grid_resolution=50)

I Got Error Message, 
if not isinstance(gbrt, BaseGradientBoosting):
        raise ValueError('gbrt has to be an instance of BaseGradientBoosting')

I Check the source code, Seems XGBClassifier is not based on BaseGradientBoosting

So the sklearn wrapper is not totally available for all feature in sklean?

**Comment by shanabraham** [comment]
I ran into the same issue. Is there a PDP function in python that works with XGBClassifier?

**Comment by yassineAlouini** [comment]
I use the following function (highly inspired from this [notebook](https://github.com/XiaoxiaoWang87/xgboost_mono_test/blob/master/xgboost_monotonicity_tutorial.ipynb): 

```
import numpy as np

def partial_dependency(model, X, features, selected_feature, floor):
    # The model could be an XGBoost sklearn fitted instance (or anything else with a 
    # predict method)
    X_temp = X.copy()
    # Works only for numerical features. Find something else for categorical features.
    grid = np.linspace(np.percentile(X_temp.loc[:, selected_feature], 0.1), 
                       np.percentile(X_temp.loc[:, selected_feature], 99.5), 
                       50)
    y_pred = np.zeros(len(grid))
    
    for i, val in enumerate(grid):
        X_temp.loc[:, selected_feature] = val
        y_pred[i] = model.predict(X_temp.loc[:, features]).mean()
    
    return grid, y_pred
```

I think it is possible to use the sklearn `plot_partial_dependence` function if you remove the 
`if not isinstance(gbrt, BaseGradientBoosting)` check.

To sum up, it would be nice to integrate a `plot_partial_dependence` into the `XGBoost` Python API.
